internal/compiler/widgets/fluent-base/widget-table-view.slint

```diff
@@ -0,0 +1,220 @@
+    min-height: 42px;
+
+    touch-area := TouchArea {}
+
+    HorizontalLayout {  
+       @children
+    }
+
+    states [
+        hover when touch-area.has-hover : {
+            background: Palette.neutralLighter;
+        }
+    ]
+}
+
+export StandardTableView := Rectangle {
+    callback column-clicked(int);
+    
+    in-out property <[TableColumn]> columns;
+    in property <[StandardTableRow]> rows;
+    in property <length> min-header-height: 42px;
+   
+    min-width: 400px;
+    min-height: 200px;
+    horizontal-stretch: 1;
+    vertical-stretch: 1;
+
+    VerticalLayout {  
+        HorizontalLayout {  
+            padding-right: 20px;
+            min-height: root.min-header-height;
+            vertical-stretch: 0;
+            for column[index] in columns : TableViewColumn {
+                ascending: column.sort-by-ascending;
+                descending: column.sort-by-descending;
+                min-width: column.min-width;
+                horizontal-stretch: column.horizontal-stretch;
+
+                Text {  
+                    vertical-alignment: center;
+                    horizontal-stretch: column.horizontal-stretch;
+                    min-width: column.min-width;
+                    text: column.title;
+                    font-weight: 900;
+                }
+                
+                clicked => {  
+                    column_clicked(index);
+                }
+            }
+        }
+
+        scroll-view := ScrollView {  
+            border-width: 0;
+            vertical-stretch: 1;
+
+            VerticalLayout {  
+                alignment: start;
+
+                for row in rows : TableViewRow {
+                    for cell[index] in row.items : TableViewCell {
+                        private property <bool> has_inner_focus;
+                        horizontal-stretch: columns[index].horizontal-stretch;
+                        min-width: columns[index].min-width;
+
+                        Rectangle {  
+                            // height: 100%;
+                            line-edit := LineEditInner {
+                                opacity: has-focus ? 1.0 : 0.0;
+                                visible: cell.editable;
+                                text: cell.text;
+                                edited => {  
+                                    cell.text = text;
+                                }
+                            }
+    
+                            if(!cell.editable || !line-edit.has-focus) : Text {
+                                width: 100%;
+                                height: 100%;
+                                overflow: elide;
+                                vertical-alignment: center;
+                                text: cell.text;
+                            }
+                        }
+                    }
+                }
+            }
+           
+        } 
+    }
+}
+
+TableViewTest := Window {
```

Comment:
I guess you plan to remove that, or move it in tests/cases

Reply:
It's removed. I think it would make sense to implement a dedicated test after we agreed on the api.